### PR TITLE
[CARBONDATA-2094] Filter DataMap Tables in Show Table Command

### PR DESCRIPTION
**Pull request by BJangir**
Currently Show Table command shows datamap tables (agg tablels) but show table command should not show aggregate tables.

Solution :- Handle "show table "command in carbon side and Filter the datamap table and return rest of  the tables.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? NO
 
 - [ ] Any backward compatibility impacted?
 NO
 - [ ] Document update required?
NO
 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
UI Added
        - How it is tested? Please attach test report.
UI added
        - Is it a performance related change? Please attach the performance test report.
 NA        
- Any additional information to help reviewers in testing this change.
      NA 
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2002/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3236/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3192/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import scala.collection.JavaConverters._
+
+import org.apache.spark.sql.{CarbonEnv, Row, SparkSession}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference}
+import org.apache.spark.sql.execution.command.MetadataCommand
+import org.apache.spark.sql.types.{BooleanType, StringType}
+
+
+private[sql] case class ShowTablesCommand ( databaseName: Option[String],
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala]: Change to `CarbonShowTablesCommand` like other commands

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    // instead of calling tables in sparkSession.
+    // filterDataMaps Method is to Filter the Table.
+    val catalog = sparkSession.sessionState.catalog
+    val db = databaseName.getOrElse(catalog.getCurrentDatabase)
+    var tables =
+      tableIdentifierPattern.map(catalog.listTables(db, _)).getOrElse(catalog.listTables(db))
+    tables = filterDataMaps(tables, sparkSession)
+    tables.map { tableIdent =>
+      val isTemp = catalog.isTemporaryTable(tableIdent)
+      Row(tableIdent.database.getOrElse(""), tableIdent.table, isTemp)
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala]: `tableIdent.database.getOrElse("")` is not right, default database is `default`

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }
+
+  private def filterDataMaps(tables: Seq[TableIdentifier],
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala]: please add comment

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      sparkSession: SparkSession): Seq[TableIdentifier] = {
+    // Filter carbon Tables then get CarbonTable and getDataMap List and filter the same
+    // as of now 2 times lookup is happening(filter  carbon table ,getDataMapList)
+    // <todo> add another PR to improve  with 1 lookup
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala]: change <todo> to TODO
and add the JIRA number here

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    val allDatamapTable = tables.filter(tbs => {
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala]: `tables.filter(tbs => {` change to 
`tables.filter { table =>`

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala (hunk 1 above)
Comment (BJangir) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/ShowTablesCommand.scala]: OK

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3294/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2056/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3331/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2095/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonShowTablesCommand.scala [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+   *
+   * @param tables tableIdnetifers
+   * @param sparkSession sparksession
+   * @return  Tables after filter datamap tables
+   */
+  private def filterDataMaps(tables: Seq[TableIdentifier],
+      sparkSession: SparkSession): Seq[TableIdentifier] = {
+    // Filter carbon Tables then get CarbonTable and getDataMap List and filter the same
+    // as of now 2 times lookup is happening(filter  carbon table ,getDataMapList)
+    // <todo> add another PR (CARBONDATA-2103) to improve  with 1 lookup
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonShowTablesCommand.scala]: change to `TODO`

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonShowTablesCommand.scala [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    val allDatamapTable = tables.filter(table => {
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonShowTablesCommand.scala]: the code style is not changed yet

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3364/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3258/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2128/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3277/

**Comment by kumarvishal09** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2176/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3413/

**Comment by kumarvishal09** [comment]
LGTM

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggregateDrop.scala [hunk 8]
```diff
@@ -84,7 +86,8 @@ class TestPreAggregateDrop extends QueryTest with BeforeAndAfterAll {
     sql("create datamap preagg_same1 on table maintable using 'preaggregate' as select" +
         " a,sum(c) from maintable group by a")
     showTables = sql("show tables")
```
Comment (xubo245) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggregateDrop.scala]: remove this line

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggCreateCommand.scala [hunk 9]
```diff
@@ -270,6 +284,28 @@ class TestPreAggCreateCommand extends QueryTest with BeforeAndAfterAll {
...
+    checkExistence(sql("show tables"), false, "tbl_1_agg1")
+    checkExistence(sql("show tables"), true, "sparktable","tbl_1")
+  }
+
+
+  test("remove TimeSeries agg tables from show table command") {
+    sql("DROP TABLE IF EXISTS tbl_1")
+    sql("create table if not exists  tbl_1(imei string,age int,mac string ,prodate timestamp,update timestamp,gamepoint double,contrid double) stored by 'carbondata' ")
+    sql(
+      "create datamap agg2 on table tbl_1 using 'preaggregate' DMPROPERTIES ('timeseries" +
```
Comment (xubo245) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggCreateCommand.scala]: we don't support create timeseries table like this :  'timeseries.eventTime'='prodate', 'timeseries.hierarchy'

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggCreateCommand.scala [hunk 10]
```diff
@@ -216,6 +216,20 @@ class TestPreAggCreateCommand extends QueryTest with BeforeAndAfterAll {
...
+    sql("DROP TABLE IF EXISTS tbl_1")
+    sql("DROP TABLE IF EXISTS sparktable")
+    sql("create table if not exists  tbl_1(imei string,age int,mac string ,prodate timestamp,update timestamp,gamepoint double,contrid double) stored by 'carbondata' ")
+    sql("create table if not exists sparktable(a int,b string)")
+    sql(
+      s"""create datamap preagg_sum on table tbl_1 using 'preaggregate' as select mac,avg(age) from tbl_1 group by mac"""
+        .stripMargin)
+    sql(
+      "create datamap agg2 on table tbl_1 using 'preaggregate' DMPROPERTIES ('timeseries" +
+      ".eventTime'='prodate', 'timeseries.hierarchy'='hour=1,day=1,month=1,year=1') as select prodate," +
```
Comment (xubo245) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggCreateCommand.scala]: we don't support create timeseries table like this now : 'timeseries.eventTime'='prodate', 'timeseries.hierarchy'

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggregateDrop.scala [hunk 11]
```diff
@@ -46,8 +46,9 @@ class TestPreAggregateDrop extends QueryTest with BeforeAndAfterAll {
       " a,sum(c) from maintable group by a")
     sql("drop datamap if exists preagg2 on table maintable")
     val showTables = sql("show tables")
+    val showdatamaps =sql("show datamap on table maintable")
     checkExistence(showTables, false, "maintable_preagg2")
```
Comment (xubo245) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggregateDrop.scala]: this test case is invalid after you change the function of "show tables

**Comment by xubo245** [comment]
There are many test case is unuseful/invalid after you change the function of  "show tables"

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggregateDrop.scala (hunk 11 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggregateDrop.scala]: for example:

  test("dropping 1 aggregate table should not drop others") {
     sql(
       "create datamap preagg1 on table maintable using 'preaggregate' as select" +
       " a,sum(b) from maintable group by a")
     sql(
       "create datamap preagg2 on table maintable using 'preaggregate' as select" +
       " a,sum(c) from maintable group by a")
     sql("drop datamap if exists preagg2 on table maintable")
     val showTables = sql("show tables")
     checkExistence(showTables, false, "maintable_preagg2")

showTables will always not contain maintable_preagg2, so we need adapt new show table function. There are some test case need check again.  @BJangir @jackylk @chenliang613 @sraghunandan @ravipesala

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggCreateCommand.scala (hunk 10 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggCreateCommand.scala]: I fix it on https://github.com/apache/carbondata/pull/1929, please review it @BJangir @jackylk @ravipesala @sraghunandan

**Pull request closed without merging**